BUILDER_DEV.md

```diff
@@ -1,225 +1,109 @@
 
-```
-export HAB_AUTH_TOKEN="<your github token>"
-export HAB_BLDR_URL="http://localhost:9636"
-export HAB_ORIGIN="core"
-```
+1. [Setup a GitHub application](https://github.com/settings/apps/new) for your GitHub organization.
+1. Set the value of `Homepage URL` to `http://${APP_HOSTNAME}`
+1. Set the value of `User authorization callback URL` to `http://${APP_HOSTNAME}/` (The trailing `/` is *important*)
+1. Set the value of `Webhook URL` to `http://${APP_HOSTNAME}/`
+1. Set everything to read only (this is only used for your org so it's safe)
+1. Save and download the pem key
+1. Copy the pem key to `${HABITAT_SRC_ROOT}/.secrets/builder-github-app.pem`
+1. Record the the client-id, client-secret, app_id and webhook-secret. These will be used for the `GITHUB_CLIENT_ID`, `GITHUB_CLIENT_SECRET`, `GITHUP_APP_ID` and `GITHUB_WEBHOOK_SECRET` build variables (seen below).
 
-* symlink /src to path/to/your/habitat/repo
+### Create app env file
 
+1. Create `${HABITAT_SRC_ROOT}/.secrets/habitat-dev`
+1. Add
 ```
-$ sudo ln -s path/to/your/habitat/repo /src
+  APP_HOSTNAME=localhost
+  GITHUB_API_URL=https://api.github.com
+  GITHUB_WEB_URL=https://github.com
+  GITHUB_CLIENT_ID=${GITHUB_CLIENT_ID}
+  GITHUB_CLIENT_SECRET=${GITHUB_CLIENT_SECRET}
+  GITHUB_WEBHOOK_SECRET=${GITHUB_WEBHOOK_SECRET}
+  GITHUB_APP_ID=${GITHUP_APP_ID}
 ```
+1. Save and close
 
-* Cd into your Habitat repo
+### Studio requirements
 
-```
-$ cd /src
-```
+1. Ensure you have run `hab setup` in your environment that will be executing the studio
 
-* Run the provision script
+### Starting the services
 
-```
-$ ./support/linux/provision.sh
-```
+From either your VM or Mac:
 
-* Source cargo
-
-```
-$ source $HOME/.cargo/env
-```
+* `cd <habitat source path>`
+* `direnv allow`
+* `hab studio enter`
+* `start-builder`
 
-* Run the build-srv makefile (NOTE - if you receive an error about cargo not being found, make sure you have removed the "secure_path" section of /etc/sudoers)
-
-```
-$ make build-srv
-```
-
-* Run this command (it may take awhile) to start all the Builder services
-
-```
-$ sudo -E make bldr-run-no-build
-```
+### UI Setup
 
-* Wait for all services to start - they are started when you see worker heartbeat debug entries in the log. If services do not start because some were already running from a previous attempt, run `$ make bldr-kill`
+If you are not doing UI development you just need to navigate to `${APP_HOSTNAME}/#/sign-in`
 
-### UI Setup
+If you are developing the UI:
 
 * Follow the instructions in the [Web UI README](https://github.com/habitat-sh/habitat/blob/master/components/builder-web/README.md) to get the Web UI running locally.
 * Open up a browser and navigate to http://localhost:3000/#/pkgs - you should see the Habitat UI running locally.
 
-### Creating an Origin
+## Helper functions
 
-### Option A: Create an origin using the web UI
+* `start-builder` - Run the builder cluster
+* `origin` - Create the core origin
```

Comment:
This failed for me because `${HAB_AUTH_TOKEN}` wasn't set in the studio despite inputting it when prompted by `hab setup` before entering the studio. When I ran the `curl` command directly with my token supplied, it created the origin correctly. Some combination of `set -u` in the `origin` function or adding `--fail` to the curl command would be useful. That way `origin` would output:
```
bash: HAB_AUTH_TOKEN: unbound variable
```
or
```
curl: (22) The requested URL returned error: 401 Unauthorized
```

I'm not sure if there's a way to `set -u` it just for the duration of the function, or if it would be good to have it set in the studio more generally.

Reply:
I added `set -u` at the top of the file